src/mlpack/methods/edge_boxes/feature_extraction_impl.hpp

```diff
@@ -0,0 +1,1039 @@
+/**
+ * @file feature_extraction_Impl.hpp
+ * @author Nilay Jain
+ *
+ * Implementation of feature extraction methods.
+ */
+#ifndef MLPACK_METHODS_EDGE_BOXES_STRUCTURED_TREE_ImPL_HPP
+#define MLPACK_METHODS_EDGE_BOXES_STRUCTURED_TREE_ImPL_HPP
+
+
+#include "feature_extraction.hpp"
+#include <mlpack/methods/pca/pca.hpp>
+namespace mlpack {
+namespace structured_tree {
+
+/**
+ * Constructor: stores all the parameters in an object
+ * of feature_parameters class.
+ */
+template<typename MatType, typename CubeType>
+StructuredForests<MatType, CubeType>::
+StructuredForests(FeatureParameters F) 
+{
+  params = F; 
+}
+
+/*
+template<typename MatType, typename CubeType>
+MatType StructuredForests<MatType, CubeType>::
+LoadData(MatType const &Images, MatType const &boundaries,\
+     MatType const &segmentations)
+{
+  const size_t num_Images = this->params.num_Images;
+  const size_t rowSize = this->params.RowSize();
+  const size_t colSize = this->params.ColSize();
+  MatType input_data(num_Images * rowSize * 5, colSize);
+  // we store the input data as follows: 
+  // Images (3), boundaries (1), segmentations (1).
+  size_t loop_iter = num_Images * 5;
+  size_t row_idx = 0;
+  size_t col_i = 0, col_s = 0, col_b = 0;
+  for(size_t i = 0; i < loop_iter; ++i)
+  {
+    if (i % 5 == 4)
+    {
+      input_data.submat(row_idx, 0, row_idx + rowSize - 1,\
+        colSize - 1) = MatType(segmentations.colptr(col_s),\
+                                  colSize, rowSize).t();
+      ++col_s;
+    }
+    else if (i % 5 == 3)
+    {
+      input_data.submat(row_idx, 0, row_idx + rowSize - 1,\
+        colSize - 1) = MatType(boundaries.colptr(col_b),\
+                                  colSize, rowSize).t();
+      ++col_b;
+    }
+    else
+    {
+      input_data.submat(row_idx, 0, row_idx + rowSize - 1,\
+        colSize - 1) = MatType(Images.colptr(col_i),
+                                  colSize, rowSize).t();
+      ++col_i;  
+    }
+    row_idx += rowSize;
+  }
+  return input_data;
+} 
+
+*/
+
+/**
+ * Get DImensions of Features
+ * @param FtrDim Output vector that contains the result 
+ */
+template<typename MatType, typename CubeType>
+void StructuredForests<MatType, CubeType>::
+GetFeatureDimension(arma::vec& FtrDim)
+{
+  FtrDim = arma::vec(2);
+
+  const size_t shrink = this->params.Shrink();
+  const size_t pSize = this->params.PSize();
+  const size_t numCell = this->params.NumCell();
+  const size_t numOrient = this->params.NumOrient();
+  
+  size_t nColorCh;
+  if (this->params.RGBD() == 0)
+    nColorCh = 3;
+  else
+    nColorCh = 4;
```

Comment:
Another solution(a problem of taste, pick the one you like more :) ) : 
`const size_t nColorCh = params.RGBD() == 0 ? 3 : 4;`